exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/DateTypeFunctions.java

```diff
@@ -40,6 +42,41 @@
 
 public class DateTypeFunctions {
 
+    /**
+     * Function to check if a varchar value can be cast to a date.
+     *
+     * At the time of writing this function, several other databases were checked
+     * for behavior compatibility. There was not a consensus between oracle and
+     * Sql server about the expected behavior of this function, and Postgres
+     * lacks it completely.
+     *
+     * Sql Server appears to have both a DATEFORMAT and language locale setting
+     * that can change the values accepted by this function. Oracle appears to
+     * support several formats, some of which are not mentioned in the Sql
+     * Server docs. With the lack of standardization, we decided to implement
+     * this function so that it would only consider date strings that would be
+     * accepted by the cast function as valid.
+     */
+    @SuppressWarnings("unused")
+    @FunctionTemplate(name = "isdate", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls=NullHandling.INTERNAL,
+        costCategory = FunctionTemplate.FunctionCostCategory.COMPLEX)
+    public static class CastVarCharToDate implements DrillSimpleFunc {
+
+      @Param NullableVarCharHolder in;
```

Comment:
Do we need an implementation for non-nullable input, since it's using NullHandling.INTERNAL?